lib/ruby_smb/smb2/tree.rb

```diff
@@ -60,78 +60,16 @@ def open_pipe(opts)
         # Make sure we don't modify the caller's hash options
         opts = opts.dup
         opts[:filename] = opts[:filename].dup
-        opts[:filename] = opts[:filename][1..-1] if opts[:filename].start_with? '\\'
+        opts[:filename] = opts[:filename][1..-1] if opts[:filename].start_with?('\\'.encode(opts[:filename].encoding))
         open_file(**opts)
```

Comment:
I'm wondering if this should also be a call to `_open(**opts)` instead?

Reply:
You're right I missed that. It's fixed now though.